Story Missions: The Story (#587)
ClaudeMetz/Alt-F4: submissions/story-missions/story-missions.md

Yeah, never say never. :D Suddenly from one day to the next I got insanely bored and I turned Faktorio back on in the spring of 2021. I actually originally wanted to delete the game, but instead I turned it on and looked at old saved positions. I also found the old save files from the New Hope Scenario mod (the original fan campaign from the beta testing period, which the game's authors then used as a template for the tutorial, if I'm not mistaken). I looked at the map structure again and started digging through the scenario files. I found it strange that after the patch (I think 1.1) the mod no longer appeared in the main menu section. Since my professional job is to fix everything, I wanted to find out why it wasn't working, and when I found out why and actually looked at the control.lua structure of the scenario, I thought "why not try that too". And that's where it all started.

{ gallery
media/game1.jpg
media/mission-2-intro.jpg
media/2.jpg
}

> I went through all the tutorial map files and the New Hope campaign, and through that I learned how to make my own missions. Then all it took was a bit of inspiration (Starship Troopers, Star Wars, Star Trek, Star Gate, Avatar), a bit of experience playing games like StarCraft 1-2, Warcraft 3, Age of Empires 2, Command & Conquer (Generals, Tiberium Wars, Red Alert 2, Dune 2000) and then some heavy self-criticism and voilá (that's French). :D Within 3-4 months, a campaign of 5 missions and an improvised fictional story was created, trying to be as neutral and lore-friendly to the original game as possible. Technically, the whole campaign takes place about 2 years after the engineer first crash-landed on the planet Nauvis. Somehow he survived and sent up a satellite rocket and was rescued a few months later. Subsequently, Earth decided to "colonize" the planet, taking even more resources than the engineer himself had before (RDA from Avatar :D :D ). Add to that the space thugs "i.e. pirates" :D (I always love it when you say "Space Pirate", I always think of Captain Harloc from that anime series/movie) :D, it starts to get pretty cool. I won't tell you the plot of the mod here of course, play it yourself and judge. Admittedly I'm not like J.K. Rowling (Harry Potter) and it's really hand written and also what mood I was in that time. It's safe to say that I'm not entirely happy with the story, as it's not the best I've ever put out, but it does fulfill its purpose.

Reviewer: it is "voila" in french, no fancy accents ;)

### [Story Missions](https://mods.factorio.com/mod/Story-Missions): The Story <author>GreenFlag</author>

> I want to talk about the Story Missions mod. It's actually hard to decide where to start, because the idea of additional objectives for the campaign was born in my head more than a year ago. When I first finished the entire tutorial of the base game and then played freeplay, after 200 hours I started to feel the slightly repetitive gameplay and was starting to miss a challenge. At the same time, I didn't want to keep starting a new game and building and developing everything from scratch. So I came up with the idea of making my own maps along the theme of the tutorial, only it wouldn't be a tutorial anymore, but regular missions instead.

Reviewer 2: ```suggestion
I want to talk about my Story Missions mod. It's actually hard for me to decide where to start, because the idea of additional objectives for the campaign was born in my head more than a year ago. After 200 hours of Factorio I started to feel the slightly repetitive gameplay and was starting to miss a challenge. At the same time, I didn't want to keep starting a new game and building and developing everything from scratch. So I came up with the idea of making my own maps along the theme of the tutorial, only it wouldn't be a tutorial anymore, but regular missions instead.
```

When I first tried the Factorio editor, I was surprised that it was actually quite limited. I had a lot of experience with editors for StarCraft/Warcraft, in which you would write scripts in-game, using a system based on "triggers". Basically, you have to find a good trigger which matches your requirements as closely as possible, then you have to just add values and that's it. I was sad that in Factorio, all the scripts are not able to be pre-defined in some interface directly in the game, but as pure code instead. The basis of all this are `.lua` files. After discovering this problem, and in fact taking the time to re-learn how to program after 10 years, I shut the game down for good and haven't turned it back on for a few months. (I hope no programmer will blame me here.)

{ gallery

> ![In Mission 2 bitters attacking on the Hell's Gate Space Port (This is reference on Starship Troopers and Avatar)](media/mission-2-intro.jpg)

Reviewer 2: ```suggestion
![In Mission 2 biters attacking Hell's Gate space port (This is a reference to Starship Troopers and Avatar)](media/mission-2-intro.jpg)
```

> ![The text bubbles like in tutorial and Engineer speeches to himself. This is how Story Mission mod is created.](media/game1.jpg)

Reviewer 2: ```suggestion
![Text bubbles like in the tutorial with the Engineer talking to himself. This is how the Story Missions mod works.](media/game1.jpg)
```

> ![In Mission 2 bitters attacking on the Hell's Gate Space Port (This is reference on Starship Troopers and Avatar)](media/2.png)

Reviewer 2: ```suggestion
![In Mission 2 biters attacking Hell's Gate space port (This is a reference to Starship Troopers and Avatar)](media/2.png)
```

}

Yeah, never say never. Suddenly from one day to the next I got insanely bored and I turned Factorio back on in the Spring of 2021. I actually originally wanted to delete the game, but instead I turned it on and looked at old save files from the New Hope scenario. I looked at the map structure again and started digging through the scenario files. I found it strange that after the patch (I think it was 1.1), the mod no longer appeared in the main menu section. Since my professional job is to fix everything, I wanted to find out why it wasn't working. When I found out why and actually looked at the `control.lua` structure of the scenario, I thought: "Why not try that too?", and that's where it all started.

> I went through all the tutorial map files and the New Hope campaign, and through that I learned how to make my own missions. Then all it took was a bit of inspiration (Starship Troopers, Star Wars, Star Trek, Star Gate, Avatar), a bit of experience playing games like StarCraft 1-2, Warcraft 3, Age of Empires 2, Command & Conquer (Generals, Tiberium Wars, Red Alert 2, Dune 2000) and then some heavy self-criticism, et voilà (that's French): Within 3-4 months, a campaign of five missions and an improvised fictional story was created, trying to be as neutral and lore-friendly to the original game as possible.

Reviewer 2: ```suggestion
I went through all the tutorial map files and the New Hope campaign, and through that I learned how to make my own missions. Then all it took was a bit of inspiration (Starship Troopers, Star Wars, Star Trek, Star Gate, Avatar), a bit of experience playing games like StarCraft 1-2, Warcraft 3, Age of Empires 2, Command & Conquer (Generals, Tiberium Wars, Red Alert 2, Dune 2000) and then some heavy self-criticism, et voilà (that's French): Within 3-4 months, a campaign of five missions and an improvised story was created, trying to be as lore-friendly to the original game as possible.
```

> Technically, the whole campaign takes place about two years after the engineer first crash-landed on the planet Nauvis. Somehow, he survived and sent up a rocket with a satellite and was rescued a few months later. Subsequently, Earth decided to colonize the planet, taking even more resources than the engineer himself had beforehand! Add to that the space thugs, i.e. "pirates" (I love it when you say "Space Pirate", I always think of anime series/movie Captain Harlock), it starts to get pretty cool. I won't tell you the plot of the mod here of course, play it yourself and find out. Admittedly, I'm not J.K. Rowling and it's really hand-written, and reflects the mood I was in that time. It's safe to say that I'm not entirely happy with the story, as it's not the best I've ever put out, but it does fulfill its purpose.

Reviewer 2: ```suggestion
The whole campaign takes place about two years after the engineer first crash-landed on the planet Nauvis. Somehow, he survived and sent up a rocket with a satellite and was rescued a few months later. Subsequently, Earth decided to colonize the planet, taking even more resources than the engineer himself had beforehand! Add to that the space thugs, i.e. "pirates" (I love it when you say "Space Pirate", I always think of anime series/movie Captain Harlock), it starts to get pretty cool. I won't tell you the plot of the mod here of course, you'll have to play it yourself and find out. Admittedly, I'm not J.K. Rowling, and the plot reflects the mood I was in at the time. It's safe to say that I'm not entirely happy with the story, as it's not the best I've ever put out, but it does fulfill its purpose.
```

> Well, I guess we're at the end of the "how the Story Missions mod came to be" story. I could go on and on about the creation of each mission, but you'll find that as an Easter egg at the end of the fifth mission (when you hit continue after completing the game). As far as I can tell, the story so far is not finished and I am currently working on five more missions to continue the adventure. I have to say that if you play mission 9, you will all be very pleasantly surprised (if not cheering with joy), because I am already super excited about it as well. I'll give you a little hand-made teaser here.

Reviewer 2: ```suggestion
Well, I guess we're at the end of the "how the Story Missions mod came to be" story. I could go on and on about the creation of each mission, but you'll find that as an Easter egg at the end of the fifth mission (when you hit continue after completing the game). The story so far is not finished and I am currently working on five more missions that continue the adventure. You should all be pleasantly surprised by mission 9 (or even cheering with joy). I'll give you a little teaser:
```

{ gallery

> ![Mission 6 from upcoming mod update. Yes, you will finally get logistic robots to control!](media/teaser-mission-6.jpg)

Reviewer 2: ```suggestion
![Mission 6 from the upcoming mod update. Yes, you will finally get logistic robots to control!](media/teaser-mission-6.jpg)
```

![Scripted atomic bombardment test from upcoming mod update. Checking how game handle multiple explosions at once.](media/teaser-mission-8.jpg)

> ![Mission 9 from upcoming mod update. If your are true fan of the Factorio, guess what is this. You can twice!](media/teaser-mission-9.jpg)

Reviewer 2: ```suggestion
![Mission 9 from the upcoming mod update. If your are a true fan of Factorio, guess what this is. (You can guess twice!)](media/teaser-mission-9.jpg)
```

}

I say goodbye and thank you all for your support.

> P.S. It’s weird to me that Factorio is a Czech game and I’m citizen of Czech Republic. It would be nice to some day meet the autors of the game :D

Reviewer 2: ```suggestion
P.S. It’s weird to me that Factorio is a Czech game and I’m citizen of Czech Republic. It would be nice to someday meet the authors of the game :D
```
